src/Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/ArtworkDetails.tsx

```diff
@@ -61,34 +73,45 @@ export const ArtworkDetails: React.FC<ArtworkDetailsProps> = ({
       : undefined
 
     if (relayEnvironment) {
+      let submissionData: SubmissionInput
+      submissionData = {
+        externalId: submission?.externalId,
+        artistID: artworkDetailsForm.artistId,
+        year: artworkDetailsForm.year,
+        title: artworkDetailsForm.title,
+        medium: artworkDetailsForm.materials,
+        attributionClass: artworkDetailsForm.rarity
+          .replace(" ", "_")
+          .toUpperCase() as ConsignmentAttributionClass,
+        editionNumber: artworkDetailsForm.editionNumber,
+        editionSizeFormatted: artworkDetailsForm.editionSize,
+        height: artworkDetailsForm.height,
+        width: artworkDetailsForm.width,
+        depth: artworkDetailsForm.depth,
+        dimensionsMetric: artworkDetailsForm.units,
+        provenance: artworkDetailsForm.provenance,
+        locationCity: artworkDetailsForm.location.city.trim(),
+        locationCountry: artworkDetailsForm.location.country?.trim(),
+        locationState: artworkDetailsForm.location.state?.trim(),
+        locationCountryCode: artworkDetailsForm.location.countryCode?.trim(),
+        locationPostalCode: artworkDetailsForm.postalCode?.trim() || null,
+        state: "DRAFT",
+        utmMedium: utmParams?.utmMedium,
+        utmSource: utmParams?.utmSource,
+        utmTerm: utmParams?.utmTerm,
+        sessionID: !isLoggedIn ? getENV("SESSION_ID") : undefined,
+      }
+      if (artworkId && !match?.params?.id) {
+        // @ts-ignore
+        ;(submissionData.source as ConsignmentSubmissionSource) = "MY_COLLECTION"
+        // @ts-ignore
+        ;(submissionData.myCollectionArtworkID as string) = artworkId
```

Comment:
**suggestion:** Maybe we can get rid of the `@ts-ignore` comments by changing the code to:

```suggestion
        ;(submissionData as CreateSubmissionMutationInput).source =
          "MY_COLLECTION"
        ;(submissionData as CreateSubmissionMutationInput).myCollectionArtworkID = artworkId
```